src/mavsdk/core/mavsdk_impl.cpp

```diff
@@ -595,6 +721,12 @@ void MavsdkImpl::process_user_callbacks_thread()
 
 void MavsdkImpl::start_sending_heartbeats()
 {
+    // Before sending out first heartbeats we need to make sure we have a
+    // default server component.
```

Comment:
Do we? What if a client component wants to initiate the connection? I sometimes use that with e.g. `udp://127.0.0.1:<some_port>` where the drone/mavlink-router listens on <some_port>

Reply:
So without having/being a component, you can't send a heartbeat because what would be the compid that you send it from? So, I think this is correct.

And this is indeed used in the case you mention, I think.